version: "3"
#Temporary docker file to setup services after mishap
services:
  pihole:
    container_name: pihole
    image: pihole/pihole:latest
    network_mode: host
    dns:
      - 1.1.1.1 
    environment:
      TZ: 'Europe/Warsaw'
      WEBPASSWORD: 'aCEJtadyWSoG3V'
      VIRTUAL_HOST: 'docker.pihole'
      SERVER_IP: '192.168.0.170' 
      DNSMASQ_LISTENING: "all"
      WEB_PORT: '893'
    volumes:
      - ./pihole/etc-pihole:/etc/pihole
      - ./pihole/etc-dnsmasq.d:/etc/dnsmasq.d
    cap_add:
      - NET_ADMIN
    restart: unless-stopped

  jellyfin:
    image: jellyfin/jellyfin
    container_name: jellyfin
    network_mode: bridge
    ports:
      - "895:8096"
    volumes:
      - ./jellyfin/config:/config
      - ./jellyfin/cache:/cache
      - ./jellyfin/filmy/:/filmy
      - ./jellyfin/seriale:/seriale
      - ./jellyfin/filmy_serie:/filmy_serie
      - ./jellyfin/audiobook/:/audiobook
      - ./jellyfin/media2:/media2:ro
      - ./jellyfin/Kasety:/kasety
    restart: unless-stopped
    
  nginx:
    container_name: nginx
    image: 'jc21/nginx-proxy-manager:latest'
    network_mode: bridge
    ports:
      - '890:80' # HTTP Port
      - '8943:443' # HTTPS Port
      - '891:81' # Admin Web Port
    volumes:
      - ./nginx/data:/data
      - ./nginx/letsencrypt:/etc/letsencrypt
    restart: unless-stopped

  bitwarden:
    container_name: vault
    image: vaultwarden/server:latest
    network_mode: bridge
    volumes:
       - ./vault/vw-data:/data
    ports:
    - "892:80" #WebAccess
    restart: unless-stopped

  portainer:
    container_name: portainer
    image: portainer/portainer-ce:latest
    network_mode: bridge
    ports:
      - "8944:9443"
    volumes:
        - ./portainer/data:/data
        - /var/run/docker.sock:/var/run/docker.sock
    restart: unless-stopped
